Examples/destination_plugins/AppsFlyerDestination.swift

```diff
@@ -0,0 +1,352 @@
+        
+        func applicationDidBecomeActive(application: UIApplication) {
+            AppsFlyerLib.shared().start()
+        }
+        
+        func openURL(_ url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) {
+            AppsFlyerLib.shared().handleOpen(url, options: options)
+        }
+        
+        func receivedRemoteNotification(userInfo: [AnyHashable: Any]) {
+            AppsFlyerLib.shared().handlePushNotification(userInfo)
+        }
+    }
+    
+    public func identify(event: IdentifyEvent) -> IdentifyEvent? {
+        if let userId = event.userId, userId.count > 0 {
+            AppsFlyerLib.shared().customerUserID = userId
+        }
+        
+        if let traits = event.traits?.dictionaryValue {
+            var aFTraits: [AnyHashable: Any] = [:]
+            
+            if let email = traits["email"] as? String {
+                aFTraits["email"] = email
+            }
+            
+            if let firstName = traits["firstName"] as? String {
+                aFTraits["firstName"] = firstName
+            }
+            
+            if let lastName = traits["lastName"] as? String {
+                aFTraits["lastName"] = lastName
+            }
+            
+            if traits["currencyCode"] != nil {
+                AppsFlyerLib.shared().currencyCode = traits["currencyCode"] as? String
+            }
+            
+            AppsFlyerLib.shared().customData = aFTraits
+            
+            print("hello")
+        }
+        
+        return event
+    }
+    
+    public func track(event: TrackEvent) -> TrackEvent? {
+        
+        var properties = event.properties?.dictionaryValue
+        
+        let revenue: Double? = extractRevenue(key: "revenue", from: properties)
+        let currency: String? = extractCurrency(key: "currency", from: properties, withDefault: "USD")
+        
+        if let af_revenue = revenue, let af_currency = currency {
+            properties?["af_revenue"] = af_revenue
+            properties?["af_currency"] = af_currency
+            
+            properties?.removeValue(forKey: "revenue")
+            properties?.removeValue(forKey: "currency")
+            
+            AppsFlyerLib.shared().logEvent(event.event, withValues: properties)
+            
+        } else {
+            AppsFlyerLib.shared().logEvent(event.event, withValues: properties)
+        }
+        
+        return event
+    }
+}
+
+//MARK: - UserActivities Protocol
+
+extension AppsFlyerDestination: UserActivities {
+    func continueUserActivity(_ activity: NSUserActivity) {
+        AppsFlyerLib.shared().continue(activity, restorationHandler: nil)
+    }
+}
+
+
+//MARK: - Support methods
+
+extension AppsFlyerDestination {
+    internal func extractRevenue(key: String, from properties: [String: Any]?) -> Double? {
+        
+        guard let revenueProperty =  properties?[key] as? Double else { return nil }
+        
+        if  properties?["revenue"] is String  {
```

Comment:
Change this to `if let revenue = properties?["revenue"] as? String`, this makes it to where you don't need to explicitly unwrap below (which is a naughty)